### PR TITLE
test: upgrade scenario test

### LINKED ISSUE #277: test: upgrade scenario test [ETA 02/18/2022]
**Is your feature request related to a problem?/Why is this needed**

* We should have a test to verify that we have seamless upgrade from v1 driver to v2 driver aside from CRI recovery unit tests.

**Describe the solution you'd like in detail**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by sunpa93**
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
* tests whether proper v2 CRIs are created for resources provisioned in v1 driver after driver version upgrade

**Which issue(s) this PR fixes**:

Fixes #277 

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by sunpa93** [comment]
WIP

File: .github/workflows/upgrade_test.yml
Comment (edreed) [inline .github/workflows/upgrade_test.yml:3]: It would be good to add a daily run at least, if not on PR as well.

File: .github/workflows/upgrade_test.yml
Comment (edreed) [inline .github/workflows/upgrade_test.yml:13]: westus2 is pretty heavily used and we can't easily get more quota. Can you use another region instead (e.g. westus, westus3, eastus, westeurope)?

File: .github/workflows/upgrade_test.yml
Comment (edreed) [inline .github/workflows/upgrade_test.yml:49]: Consider using `upgradeTestOutput` for output directory.

File: .github/workflows/upgrade_test.yml
Comment (edreed) [inline .github/workflows/upgrade_test.yml:58]: This path needs to be in the `azure-cluster-up.sh` output directory, e.g. `./upgradeTestOutput/cluster-down.sh`.

File: test/upgrade/upgrade_test.go
Comment (edreed) [inline test/upgrade/upgrade_test.go:63]: Remove commented out code.

**Review by edreed — CHANGES_REQUESTED** [review]
Some changes are needed to the Github Action, but otherwise LGTM.

**Review by edreed — APPROVED** [review]

**Comment by sunpa93** [comment]
verified test success.

**Comment by sunpa93** [comment]
I will merge the changes once the github action checks all complete.

**Pull request merged**
